dvc/repo/run.py

```diff
@@ -66,6 +70,35 @@ def run(
 
     with self.state:
         if not no_exec:
+            if (stage.cmd or stage.is_import) and not stage.locked:
+                existing = [
+                    out
+                    for out in stage.outs
+                    if os.path.exists(str(out))
+                    if out.changed_cache()
+                ]
+
+                statement = (
```

Comment:
do we show it even if `existing` is empty?

Reply:
should we actually check if out exists on remote?

Reply:
will it ask about --outs-no-cache as well? usually files that are put under git control?

Reply:
> do we show it even if existing is empty?

No, we check this before prompting the user

> should we actually check if out exists on remote?

:thinking: Yes! Because `stage.remove_outs` goes to each remote and deletes the output.

> will it ask about --outs-no-cache as well? usually files that are put under git control?

Yes, it will, because the point of this message is to warn the user about DVC removing files, even if you `dvc run --outs-no-cache` it will remove the outputs to make sure that the command is the one creating the output.

Reply:
so, if I have a, let's say, a metric file and run dvc repro/dvc run - it will be asking me every time ... not sure this is the best UX ...  I don't know if there other options though

Reply:
:sweat_smile: this is what I'm facing right now, all the tests related to metrics are failing because of this assumption, @shcheklein , I'll think about it

Reply:
It is using `safe_remove` so most of the time for cached outputs(metrics included) it should be fine. For not cached ones, one might not want to lose it silently, so it makes sense to ask about it there as well. As to repro, I don't think this PR is changing anything for it at all, nor should it.

Btw, the first case when output is already cached, the referenced code doesn't make sense, because it checks only for _existance_, but if the output is already cached, we should not ask about deleting it. The related discussion was about user losing the old dvc file, for which we've reached the conclusion that we should not delete the dvc file in create() right away, but rather just overwrite it when and if the command is successful(which could probably be achieved by simply removing os.unlink() in Stage.create()).